includes/components/class-attribute.php

```diff
@@ -85,9 +85,138 @@ protected function boot() {
 
 			// Disable Jetpack search.
 			add_filter( 'jetpack_search_should_handle_query', [ $this, 'disable_jetpack_search' ], 10, 2 );
+
+			// Set search order.
+			add_filter( 'posts_orderby', [ $this, 'set_search_order' ] );
+		} else {
+
+			// Alter settings.
+			add_filter( 'hivepress/v1/settings', [ $this, 'alter_settings' ] );
 		}
 	}
 
+	/**
+	 * Sets WP search order.
+	 *
+	 * @param string $orderby Order filters.
+	 * @return string
+	 */
+	public function set_search_order( $orderby ) {
+		global $wpdb;
+
+		// Check sort option.
+		if ( isset( $_GET['_sort'] ) ) {
+			return $orderby;
+		}
+
+		foreach ( [ 'listing', 'vendor', 'request' ] as $model ) {
+
+			// Get page ID.
+			$page_id = absint( get_option( 'hp_page_' . $model . 's' ) );
+
+			// Check page.
+			if ( ! ( ( $page_id && is_page( $page_id ) ) || is_post_type_archive( 'hp_' . $model ) || ( is_tax() && strpos( get_queried_object()->taxonomy, 'hp_' . $model . '_' ) === 0 ) ) ) {
+				continue;
+			}
+
+			// Set sort param.
+			$sort_param = get_option( hp\prefix( $model . '_sorting_option' ) );
+
+			// Set sort options.
+			$order_by = '';
+			$order = '';
+
+			if ( 'title' === $sort_param ) {
+
+				// Set sort options.
+				$order_by = $wpdb->posts . '.post_title';
+				$order = 'ASC';
+			} else {
+
+				// Get sort order.
+				$sort_order = 'ASC';
+
+				if ( strpos( $sort_param, '__' ) ) {
+					list($sort_param, $sort_order) = explode( '__', $sort_param );
+				}
+
+				// Get category ID.
+				$category_id = $this->get_category_id( $model );
+
+				// Get attributes.
+				$attributes = $this->get_attributes( $model, (array) $category_id );
+
+				// Get sort attribute.
+				$sort_attribute = hp\get_array_value( $attributes, $sort_param );
+
+				if ( $sort_attribute && $sort_attribute['sortable'] ) {
+
+					// Get sort field.
+					$sort_field = hp\create_class_instance( '\HivePress\Fields\\' . $sort_attribute['edit_field']['type'], [ $sort_attribute['edit_field'] ] );
+
+					if ( $sort_field && $sort_field::get_meta( 'sortable' ) ) {
+
+						// Set sort options.
+						$order_by = $sort_param . '__order';
+						$order = strtoupper( $sort_order );
+					}
+				}
+			}
+
+			// Check sort options.
+			if ( $order_by && $order ) {
+
+				// Set sort order.
+				$orderby = $order_by . ' ' . $order;
+			}
+
+			break;
+		}
+
+		return $orderby;
+	}
+
+	/**
+	 * Alters settings.
+	 *
+	 * @param array $settings Settings configuration.
+	 * @return array
+	 */
+	public function alter_settings( $settings ) {
+		foreach ( [ 'listing', 'vendor', 'request' ] as $model ) {
+
+			// Create sort form.
+			$sort_form = hp\create_class_instance( '\HivePress\Forms\\' . $model . '_sort' );
+
+			// Check sort form.
+			if ( ! $sort_form ) {
+				continue;
+			}
+
+			// Get form fields.
+			$form_fields = $sort_form->get_fields();
```

Comment:
Variable used only once, it's ok if you do this $sort_form->get_fields()['_sort']->get_args() below since there's no condition that justifies a separate valiable anyway